File: Keywords_Extraction/extract_keywords.py
import sys
import re
from bs4 import BeautifulSoup

dictionaryFile = "stopwords.txt";


def RemoveHTMLTags(source):
    soup = BeautifulSoup(source,'lxml')
    return soup.get_text()

def ReadStopWords():
    stop_words = set()
    with open(dictionaryFile) as f:
        lines = f.readlines()
    f.close()
    for line in lines:
        line = re.sub('\r\n','',line)
        stop_words.add(line)
    return stop_words


def RemoveJunkCharacters(source):
    # here I need to double check the @ before string present in c# version
    cleaned = re.sub('[^a-zA-Z0-9 \n.:_]', '', source)
    cleaned = re.sub(r'[.:,_^\s]',' ',cleaned)
    cleaned = re.sub(r'\b\d\b','',cleaned)
    cleaned = re.sub(r'\b\w{1}\b','',cleaned )
    return cleaned

def SplitWords(source):
    return source.split(' ')

def main():
    dict = {}
    outNoStopWord = []
    outNoHTML = []

    # load stop words
    stopWords = ReadStopWords();

    # read file html
    with open(sys.argv[1]) as f:
        lines = f.readlines()

    for line in lines:
        cleanedLine = RemoveHTMLTags(line)
        cleanedLine = RemoveJunkCharacters(cleanedLine)

        if (len(cleanedLine) > 0):
            outNoHTML.append(cleanedLine)

        newLine = ''

        for subString in SplitWords(cleanedLine):
            if (len(subString) > 0):
                if (not (subString.lower() in stopWords)):
                    newLine = newLine + subString + ' '

                    if (not (subString.lower() in dict)):
                        dict[subString.lower()] = 1
                    else:
                        dict[subString.lower()] += 1;

        if (len(newLine) > 0):
            outNoStopWord.append(' '+newLine)

    # write files

    fnoHTML = open(sys.argv[1] + ".noHTML", "w")
    fnoStopWord = open(sys.argv[1] + ".noStopWord", "w")
    fKeyword = open(sys.argv[1] + ".keyword", "w")

    fnoHTML.writelines(outNoHTML)
    fnoStopWord.writelines(outNoStopWord)

    fnoHTML.close()
    fnoStopWord.close()

    # Sort by value

    # Here I should skip the first pair

    sumWords = sum(dict.values())
    freq = []

    for key, value in sorted(dict.items(), key=lambda x: x[1], reverse=True):
        freq.append(key + ', ' + str(1.0 * value / sumWords)+'\n')

    fKeyword.writelines(freq)


if __name__ == '__main__':
    main()

File: Keywords_Extraction/code and example of output from c#/extract_keywords.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Text.RegularExpressions;

namespace TextFilter
{
    class Program
    {
        static string dictionaryFile = "stopwords.txt";
        static Dictionary<string, bool> stopWords = new Dictionary<string, bool>();
        
        public static string RemoveHTMLTags(string source) {
            return Regex.Replace(source, "<.*?>", string.Empty);
        }

        public static void ReadStopWords() {
            string[] lines = System.IO.File.ReadAllLines(dictionaryFile);
            foreach (string line in lines) {
                stopWords.Add(line, true);
            }
        }

        public static string RemoveJunkCharacters(string source) {
            string removedJunkChars = Regex.Replace(source, @"\b\d\b", string.Empty);
            removedJunkChars = Regex.Replace(removedJunkChars, "[^a-zA-Z0-9 .:_]", string.Empty);
            removedJunkChars = Regex.Replace(removedJunkChars, @"^\s+", string.Empty);
            return Regex.Replace(removedJunkChars, @"RawTextenglishText_\d+_\d+", string.Empty);
        }

        public static string[] SplitWords(string source) {
            return Regex.Split(source, "[^a-zA-Z0-9]+");
        }

        static void Main(string[] args) {
            ReadStopWords();
            Dictionary<string, int> dict = new Dictionary<string, int>();

            string[] lines = System.IO.File.ReadAllLines(args[0]);
            
            List<string> outNoStopWord = new List<string>();
            List<string> outNoHTML = new List<string>();
            foreach (string line in lines) {
                string cleanedLine = RemoveHTMLTags(line);
                cleanedLine = RemoveJunkCharacters(cleanedLine);

                if (cleanedLine.Length > 0) {
                    outNoHTML.Add(cleanedLine);
                }

                string newLine = "";
                foreach (string substring in SplitWords(cleanedLine)) {
                    if (substring.Length > 0) {
                        if (!stopWords.ContainsKey(substring.ToLower())) {
                            newLine = newLine + substring + " ";

                            if (!dict.ContainsKey(substring.ToLower())) {
                                dict.Add(substring.ToLower(), 1);
                            }
                            else {
                                dict[substring.ToLower()]++;
                            }
                        }
                    }
                }

                if (newLine.Length > 0) {
                    outNoStopWord.Add(newLine);
                }
            }
            System.IO.File.WriteAllLines(args[0] + ".noHTML", outNoHTML.ToArray());
            System.IO.File.WriteAllLines(args[0] + ".noStopWord", outNoStopWord.ToArray());

            // sort by value
            double sumWords = dict.Skip(1).Sum(v => v.Value);
            List<string> freq = new List<string>();
            foreach (KeyValuePair<string, int> item in dict.OrderByDescending(key => key.Value)) {
                freq.Add(String.Format("{0}, {1}", item.Key, 1.0 * item.Value / sumWords));
            }
            System.IO.File.WriteAllLines(args[0] + ".keyword", freq.ToArray());
        }
    }
}